docs/docs-beta/docs/guides/non-python.md

```diff
@@ -1,73 +1,84 @@
 
 ![Image of captured stdout](/img/placeholder.svg)
 
-## Send to, and receive data from, your script (Optional)
+## Step 3: Send and receive data from the script
+
+To send context to your script or emit events back to Dagster, you can use environment variables provided by the `PipesSubprocessClient`.
 
-Optionally, you may want to send context to your script, or emit events back to Dagster. This can be done through environment variables that are provided by the `PipesSubprocessClient`.
+To specify a file path, use the following environment variables:
 
-The two environment variables specify a path to a file&mdash;one for input and one for output:
+- `DAGSTER_PIPES_CONTEXT` - Input context
+- `DAGSTER_PIPES_MESSAGES` - Output context
 
-- `DAGSTER_PIPES_CONTEXT`: Input context
-- `DAGSTER_PIPES_MESSAGES`: Output context
+Create a new file with the following helper functions that read the environment variables, decode the data, and write messages back to Dagster:
 
-Create the following helper functions for reading the environment variables, decoding the data, and writing messages back to Dagster.
+<CodeExample filePath="guides/non-python/pipes-javascript-utility.js" language="javascript" />
 
-<CodeExample filePath="guides/non-python/pipes-javascript-utility.js" language="javascript" title="Utility functions to interface with Dagster Pipes." />
+Both environment variables are base64 encoded, zip compressed JSON objects. Each JSON object contains a path that indicates where to read or write data.
 
-Both environment variables are base64 encoded, zip compressed JSON objects. Each JSON object contains a _path_ that indicate where to read or write data.
+## Step 4: Integrate the utility functions into the script
 
-## Integrate the utility functions into your original script
+Using the utility functions to decode the Dagster Pipes environment variables, you can send additional parameters into the JavaScript process. You can also output more information into the asset materializations.
 
-With the utility functions to decode the Dagster Pipes environment variables, we can send additional parameters into the JavaScript process and output additional information into the asset materializations.
+Update the `tensorflow/main.js` script to:
 
-Update the `tensorflow/main.js` script to retrieve the model configuration from the Dagster context, and report an asset materialization back Dagster with model metadata.
+- Retrieve the model configuration from the Dagster context, and
+- Report an asset materialization back to Dagster with model metadata
 
-<CodeExample filePath="guides/non-python/pipes-full-featured-javascript.js" language="javascript" title="Adding a new JavaScript entrypoint for Dagster Pipes." />
+<CodeExample filePath="guides/non-python/pipes-full-featured-javascript.js" language="javascript" />
 
-## Update your Dagster asset
+## Step 5: Update the Dagster asset
```

Comment:
Suggestions for a better/more helpful title?

Reply:
What about, "Update the asset to pass extra parameters to the Javascript subprocess"